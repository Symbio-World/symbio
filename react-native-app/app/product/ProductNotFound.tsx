import * as React from 'react'
import { View, Text, Linking } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { Button } from '../ui-kit/Button'
import { Modal } from '../ui-kit/Modal'

type Props = {
  barcode?: string
}
export const ProductNotFound: React.FC<Props> = ({ barcode }) => {
  const [isPressed, setPressed] = React.useState<boolean>(false)
  return (
    <View style={[t.flex1, t.justifyCenter, t.m5]}>
      {isPressed && (
        <Modal onDismiss={() => setPressed(false)}>
          <View style={[t.flex1, t.justifyCenter, t.m5]}>
            <Text>Thanks for helping, this feature will be available soon</Text>
          </View>
        </Modal>
      )}
      <Text>
        Unfortunately we could not find any data for this barcode{' '}
        <Text>{barcode}</Text>
      </Text>
      <View style={[t.mY5]}>
        <Button
          title="Search with google"
          onPress={() =>
            Linking.openURL(`https://www.google.com/search?q=${barcode}`)
          }
        />
      </View>
      <Button title="Add Manually" onPress={() => setPressed(true)} />
    </View>
  )
}
