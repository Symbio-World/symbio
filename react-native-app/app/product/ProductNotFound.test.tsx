import * as React from 'react'
import { render } from 'react-native-testing-library'
import { ProductNotFound } from './ProductNotFound'

describe('ProductNotFound', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<ProductNotFound />)
    expect(toJSON()).toMatchSnapshot()
  })
})
