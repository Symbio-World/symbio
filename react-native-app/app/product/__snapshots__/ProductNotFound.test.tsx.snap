// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductNotFound renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "margin": 20,
      },
    ]
  }
>
  <Text>
    Unfortunately we could not find any data for this barcode
     
    <Text />
  </Text>
  <View
    style={
      Array [
        Object {
          "marginVertical": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#B35340",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "fontSize": 24,
            },
            Object {
              "color": "#fff",
            },
            Array [
              Object {
                "color": "#fff",
              },
            ],
          ]
        }
      >
        Search with google
      </Text>
    </View>
  </View>
</View>
`;
